### PR TITLE
Search: match Jetpack Cloud look and feel

### PR DESCRIPTION
**Pull request by bluefuton**
#### Changes proposed in this Pull Request

The Jetpack Search section was originally only shown in Calypso blue, so exclusively uses styles that look like regular Calypso. This looks out-of-place when rendered in Jetpack Cloud.

This PR updates the section to use the Jetpack Upsell component in Cloud, which matches the style of the Backup and Scan sections. We have preserved the existing look and feel in Calypso blue.

#### Testing instructions

1. Run locally with `CALYPSO_ENV=jetpack-cloud-development yarn start` and map jetpack.cloud.localhost to 127.0.0.1 in your “hosts” file.
2. Visit http://jetpack.cloud.localhost:3000/ and choose the 'Search' item in the sidebar.
3. Ensure the appearance of the section matches other Jetpack sections like Backup and Scan (no container box, correct fonts and sizes).

Screen Shot 2021-06-02 at 12 01 11

4. Stop Calypso and restart with the regular environment (yarn start) and access via http://calypso.localhost:3000.
5. Ensure the appearance of the section matches other Jetpack sections like Backup and Scan (container box, correct fonts and sizes).

Screen Shot 2021-06-02 at 16 43 07

### CODE REVIEW

**Comment by kangzj** [comment]
Looks good :-)

The font size of paragraph in backup is 1 rem and Jetpack Search has a slightly bigger size. Maybe change it if it's not intended.
image


And also should we add a border?
image

**Comment by bluefuton** [comment]
@kangzj I'm aiming to match the appearance of Backup and Scan as those sections are pretty similar content-wise:

Screen Shot 2021-06-03 at 12 12 46

(We could do with a better logo though!)

File: client/my-sites/jetpack-search/placeholder.tsx [hunk 1]
```diff
@@ -35,17 +30,14 @@ export default function JetpackSearchPlaceholder( { siteId }: Props ): ReactElem
...
 			<SidebarNavigation />
 
-			<FormattedHeader
-				headerText={ translate( 'Jetpack Search' ) }
-				id="jetpack-search-header"
-				align="left"
-				brandFont
+			<JetpackSearchContent
+				headerText={ translate( 'Finely-tuned search for your site.' ) }
+				bodyText={ translate( 'Your visitors are getting our fastest search experience.' ) }
```
Comment (kangzj) [inline client/my-sites/jetpack-search/placeholder.tsx]: The bodyText, headerText and buttonText here don't seem to match?

File: client/my-sites/jetpack-search/placeholder.tsx [hunk 2]
```diff
@@ -35,17 +30,14 @@ export default function JetpackSearchPlaceholder( { siteId }: Props ): ReactElem
...
+				buttonLink={ null }
```
Comment (kangzj) [inline client/my-sites/jetpack-search/placeholder.tsx]: `buttonLink` is typed `TranslateResult` or `undefined`, a `null` here shows a warning on my editor. And the same case is with `onClick` too.

There are some cases other places as well, not sure if this is something we should worry about.

**Review by kangzj — COMMENTED** [review]
It works well 👍. Added comments for the parts I have questions.

File: client/my-sites/jetpack-search/placeholder.tsx (hunk 1 above)
Comment (bluefuton) [reply client/my-sites/jetpack-search/placeholder.tsx]: This is just dummy text and it's never shown to the user (hidden behind placeholder while loading). I'll make that clearer in the file.

**Review by kangzj — APPROVED** [review]
It works well and looks good to me 👍 
I restarted E2E tests several times, all failed but doesn't seem related.

**Pull request merged**